File: 08-Testing-Java/Solution/test/testing/tests/MyClassTests.java
package testing.tests;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import testing.MyClass;

class MyClassTests {

    private MyClass testee;
    private FakeLogger fakeLogger;

    @BeforeEach
    void init() {
        fakeLogger = new FakeLogger();
        testee = new MyClass(fakeLogger);
    }

    @Test
    void doLoop_LogsMessages() {
        testee.doLoop();

        Assertions.assertArrayEquals(new String[] {
                "Before loop",
                        "In loop: 1",
                        "In loop: 2",
                        "In loop: 3",
                        "After loop"},
                fakeLogger.getLoggedMessages());
    }
}

File: 08-Testing-CSharp/Solution/Testing.Tests/MyClassTests.cs
namespace Testing.Tests;

using FluentAssertions;

public class MyClassTests
{
    private readonly MyClass _testee;
    private readonly FakeLogger _fakeLogger;

    public MyClassTests()
    {
        _fakeLogger = new FakeLogger();
        _testee = new MyClass(_fakeLogger);
    }

    [Fact]
    public void DoLoop_LogsMessages()
    {
        _testee.DoLoop();

        _fakeLogger.LoggedMessages.Should().BeEquivalentTo(
            "Before loop",
            "In loop: 1",
            "In loop: 2",
            "In loop: 3",
            "After loop");
    }
}